File: models/pontos.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Pontos = sequelize.define('Pontos', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  id_user: {
    type: DataTypes.INTEGER,
    allowNull: false,
    unique: false,
  },
  vida: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
  inspiracao: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
  forca: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
  destreza: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
  constituicao: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
  inteligencia: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
  sabedoria: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
  carisma: {
    type: DataTypes.INTEGER,
    allowNull: true,
  }
}, {
  timestamps: true,  // Cria automaticamente createdAt e updatedAt
  tableName: 'pontos',  // Nome da tabela no banco
});

module.exports = Pontos;

File: models/pontos.go
package models
import (
	"time"
)
type Pontos struct {
	ID          uint   `gorm:"primaryKey;autoIncrement"`
	IDUser      uint   `gorm:"not null"`
	Vida        int
	Inspiracao  int
	Forca       int
	Destreza    int
	Constituicao int
	Inteligencia int
	Sabedoria    int
	Carisma      int
	CreatedAt time.Time `gorm:"autoCreateTime;not null"` // Campo auto gerado
	UpdatedAt time.Time `gorm:"autoUpdateTime;not null"`

}
